src/data/dfinsupp/basic.lean

```diff
@@ -148,6 +148,14 @@ end⟩
   zip_with f hf g₁ g₂ i = f i (g₁ i) (g₂ i) :=
 rfl
 
+/-- `x.piecewise y s` is the finitely supported function equal to `x` on the set `s`,
+  and to `y` on its complement. -/
+def piecewise (x y : Π₀ i, β i) (s : set ι) [Π i, decidable (i ∈ s)] : Π₀ i, β i :=
+zip_with (λ i x y, if i ∈ s then x else y) (λ _, if_t_t _ 0) x y
+
+lemma piecewise_apply (x y : Π₀ i, β i) (s : set ι) [Π i, decidable (i ∈ s)] (i : ι) :
```

Comment:
Can you add a `coe_piecewise` lemma in terms of `set.piecewise` too?

Reply:
and tag it `norm_cast`